vald/evm/evm.go

```diff
@@ -259,79 +260,49 @@ func (mgr Mgr) ProcessGatewayTxConfirmation(event *types.ConfirmGatewayTxStarted
 		return err
 	}
 
-	var events []types.Event
-	for i, log := range txReceipt.Logs {
-		if !bytes.Equal(event.GatewayAddress.Bytes(), log.Address.Bytes()) {
-			continue
-		}
+	events := processGatewayTxLogs(mgr.logger, event.Chain, event.GatewayAddress, txReceipt.Logs)
+	mgr.logger.Info(fmt.Sprintf("broadcasting vote %v for poll %s", events, event.PollID.String()))
+	_, err = mgr.broadcaster.Broadcast(context.TODO(), voteTypes.NewVoteRequest(mgr.proxy, event.PollID, types.NewVoteEvents(event.Chain, events...)))
 
-		switch log.Topics[0] {
-		case ContractCallSig:
-			gatewayEvent, err := DecodeEventContractCall(log)
-			if err != nil {
-				mgr.logger.Debug(sdkerrors.Wrap(err, "decode event ContractCall failed").Error())
-				continue
-			}
+	return err
+}
 
-			if err := gatewayEvent.ValidateBasic(); err != nil {
-				mgr.logger.Debug(sdkerrors.Wrap(err, "invalid event ContractCall").Error())
-				continue
-			}
+// ProcessGatewayTxsConfirmation votes on the correctness of an EVM chain gateway's transactions
+func (mgr Mgr) ProcessGatewayTxsConfirmation(event *types.ConfirmGatewayTxsStarted) error {
+	if !slices.Any(event.Participants, func(v sdk.ValAddress) bool { return v.Equals(mgr.validator) }) {
+		mgr.logger.Debug("ignoring gateway txs confirmation polls: not a participant", "polls", event.PollMappings)
+		return nil
+	}
 
-			events = append(events, types.Event{
-				Chain: event.Chain,
-				TxID:  event.TxID,
-				Index: uint64(i),
-				Event: &types.Event_ContractCall{
-					ContractCall: &gatewayEvent,
-				},
-			})
-		case ContractCallWithTokenSig:
-			gatewayEvent, err := DecodeEventContractCallWithToken(log)
-			if err != nil {
-				mgr.logger.Debug(sdkerrors.Wrap(err, "decode event ContractCallWithToken failed").Error())
-				continue
-			}
+	txIDs := slices.Map(event.PollMappings, func(poll types.PollMapping) common.Hash { return common.Hash(poll.TxID) })
+	txReceipts, err := mgr.GetTxReceiptsIfFinalized(event.Chain, txIDs, event.ConfirmationHeight)
+	if err != nil {
+		return err
+	}
```

Comment:
Do we want to fall back to processing tx receipts individually if this fails? Maybe some node doesn't support batching or has a max batch size or some such.

Reply:
nodes complies to spec should support batching request. 
I don't see a max batch size in spec, but true, some providers may set their own rules.

Reply:
Unless we have evidence that this could really become an issue I don't think we need to spend extra time on this. We can always still call it on single txs instead of batched